src/java/org/apache/cassandra/index/sai/disk/v3/FileValidation.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v3;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.zip.CRC32;
+
+import org.apache.cassandra.index.sai.disk.io.RAMIndexOutput;
+import org.apache.lucene.store.ByteArrayIndexInput;
+import org.apache.lucene.store.IndexInput;
+
+/**
+ * Uses CRC32 to validate the first and last 1k of a given section of a file.
+ * If the file is less than 2k then the whole file is validated.
+ */
```

Comment:
> Doing a CRC32 check was an attempt to remove the header footer check from Lucene for something different. The goal is not to check the entire file which would be slow on index load, rather it's a quick corruption check.

This would make a fantastic comment!

Reply:
The javadoc comment is pretty clear on how this class works and what it does.  Performance is not an issue here given it's 1k or 2k.  There is other code that does full file validation.